enumerable.rb

```diff
@@ -0,0 +1,349 @@
+# rubocop:disable all
+module Enumerable
+  def my_each
+    unless block_given?
+      to_enum(__method__)
+    else
+      i = 0
+      while i < self.size
+        yield self[i]
+        i += 1
+      end
+      self
+    end
+  end
+
+  def my_each_with_index
+    unless block_given?
+      to_enum(__method__)
+    else
+      i = 0
+      while i < self.size
+        yield(self[i], i)
+        i += 1
+      end
+      self
+    end
+  end
+
+  def my_select
+    unless block_given?
+      to_enum(__method__)
+    else
+      i = 0
+      arr = []
+      while i < self.size
+        yield self[i]
+        if yield self[i]
+          arr << self[i]
+        end
+        i += 1
+      end
+      arr
+    end
+  end
+
+  def my_all?(word = nil)
+    unless block_given?
+      i = 0
+      j = 0
+      k = 0
+      while i < self.size
+        self[i]
+        i += 1
+      end
+      if self.size == 0
+        true
+      elsif word.class == Regexp
+
+        for element in self
+          if element.match(word)
+            j += 1
+          end
+        end
```

Comment:
* [ ] should have working `my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5].my_all?(Integer) != [1,2,3,4,5].all?(Integer)
```